extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -350,9 +350,10 @@ private boolean shouldAutoRedirect(TenantConfigContext configContext, RoutingCon
                                 ? configContext.oidcConfig.getAuthentication().scopes.get()
                                 : Collections.emptyList();
                         List<String> scopes = new ArrayList<>(oidcConfigScopes.size() + 1);
-                        scopes.add("openid");
+                        if (configContext.oidcConfig.getAuthentication().addOpenidScope.orElse(true)) {
+                            scopes.add("openid");
+                        }
                         scopes.addAll(oidcConfigScopes);
-                        configContext.oidcConfig.getAuthentication().scopes.ifPresent(scopes::addAll);
```

Comment:
@FroMage I'm not sure why, as it does look like it duplicates the previous line, but apparently removing it cause a test failure, can you check it please

Reply:
Hmm, passed for me

Reply:
@FroMage Your code is perfectly fine, the test was broken a bit (well, I did not realize till now the scopes were duplicated - this should never a problem but is obviously redundant and can indeed cause some problems, thanks)